st2common/st2common/models/db/execution.py

```diff
@@ -61,29 +51,39 @@ class ActionExecutionDB(stormbase.StormFoundationDB):
     end_timestamp = ComplexDateTimeField(
         help_text="The timestamp when the liveaction has finished."
     )
-    parameters = stormbase.EscapedDynamicField(
+    action = stormbase.EscapedDictField(required=True)
+    parameters = JSONDictEscapedFieldCompatibilityField(
         default={},
         help_text="The key-value pairs passed as to the action runner & action.",
     )
     result = JSONDictEscapedFieldCompatibilityField(
         default={}, help_text="Action defined result."
     )
-    result_size = me.IntField(default=0, help_text="Serialized result size in bytes")
     context = me.DictField(
         default={}, help_text="Contextual information on the action execution."
     )
+    delay = me.IntField(min_value=0)
+
+    # diff from liveaction
+    runner = stormbase.EscapedDictField(required=True)
+    trigger = stormbase.EscapedDictField()
+    trigger_type = stormbase.EscapedDictField()
+    trigger_instance = stormbase.EscapedDictField()
+    rule = stormbase.EscapedDictField()
+    result_size = me.IntField(default=0, help_text="Serialized result size in bytes")
```

Comment:
todo: This is currently being set to the zstandard compressed size not the serialized size.  I need to instead set this to the  bytes  size before compression

Reply:
This should be resolved now.